### PR TITLE
OT275 76: Endpoint para listar Comentarios

### PR DESCRIPTION
**Pull request by Sieglind**
# [OT275-76](https://alkemy-labs.atlassian.net/browse/Ot275-76)

Fixed an issue related to the "CommentEntity" class, where it had a missing attribute ("body"). Created and endpoint to list all comments ordered by creation date

### HOW HAS THIS BEEN TESTED?

Tested with a Postman request, with verifiable results in the evidence provided

- [x] Postman.
- [ ] Unit test.
- [ ] Integration test.
- [ ] No testing required (only applies for tech task).

**Test configuration:**

No changes were needed to test the endpoint

**Screenshots**:
Given these comments in the database:
image

They are listed once a GET request is sent to the corresponding enpoint:
image

### CHECKLIST

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added the new resource in the Postman Collection file.

### CODE REVIEW

File: src/main/java/com/alkemy/ong/infrastructure/rest/mapper/ListCommentMapper.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.stereotype.Component;
+
+@Component
+@AllArgsConstructor
+public class ListCommentMapper {
+
+  CommentMapper commentMapper;
+
+  public ListCommentResponse toResponse(List<Comment> comments) {
+    if (comments.isEmpty()) {
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/rest/mapper/ListCommentMapper.java:17]: Falta chequear por null.

File: src/main/java/com/alkemy/ong/infrastructure/database/mapper/CommentEntityMapper.java [hunk 2]
```diff
@@ -15,7 +18,18 @@ public Comment toDomain(CommentEntity entity) {
     Comment comment = new Comment();
     comment.setId(entity.getCommentId());
     comment.setCreatedBy(userEntityMapper.toDomain(entity.getUser()));
+    comment.setBody(entity.getBody());
     return comment;
   }
 
+  public List<Comment> toDomain(List<CommentEntity> commentEntities) {
+    if (commentEntities.isEmpty()) {
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/database/mapper/CommentEntityMapper.java:26]: Falta chequear por null.

File: src/main/java/com/alkemy/ong/application/service/usecase/IListCommentUseCase.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.alkemy.ong.application.service.usecase;
+
+import com.alkemy.ong.domain.Comment;
+import java.util.List;
+
+public interface IListCommentUseCase {
+  List<Comment> findAllOrdered();
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/application/service/usecase/IListCommentUseCase.java:7]: findAllOrdered no es un nombre muy descriptivo, ya que tengo que ir a la implementación para saber porque esta ordenado.

File: src/main/java/com/alkemy/ong/application/service/ListCommentUserCaseService.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import com.alkemy.ong.application.repository.ICommentRepository;
+import com.alkemy.ong.application.service.usecase.IListCommentUseCase;
+import com.alkemy.ong.domain.Comment;
+import java.util.List;
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+public class ListCommentUserCaseService implements IListCommentUseCase {
+
+  ICommentRepository commentRepository;
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/application/service/ListCommentUserCaseService.java:12]: ENCAPSULACION & INMUTABILIDAD

File: src/main/java/com/alkemy/ong/application/repository/ICommentRepository.java [hunk 5]
```diff
@@ -12,4 +13,5 @@ public interface ICommentRepository {
 
   Optional<Comment> find(Identifiable<Long> identifiable);
 
+  List<Comment> findAllOrdered();
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/application/repository/ICommentRepository.java:16]: findAllOrdered no es un nombre muy descriptivo, ya que tengo que ir a la implementación para saber porque esta ordenado.

**Review by mkain0 — CHANGES_REQUESTED** [review]

File: src/main/java/com/alkemy/ong/infrastructure/rest/mapper/ListCommentMapper.java (hunk 1 above)
Comment (Sieglind) [reply src/main/java/com/alkemy/ong/infrastructure/rest/mapper/ListCommentMapper.java:17]: Puede retornar null el método findAll? Tenia entendido que retorna una lista vacía si no encuentra resultados

Comment (mkain0) [reply src/main/java/com/alkemy/ong/infrastructure/rest/mapper/ListCommentMapper.java]: findAll de JPA nunca retorna null, pero vos no sabes quienes pueden ser los consumers de esta clase.

File: src/main/java/com/alkemy/ong/infrastructure/database/mapper/CommentEntityMapper.java [hunk 6]
```diff
@@ -15,7 +18,18 @@ public Comment toDomain(CommentEntity entity) {
...
     comment.setCreatedBy(userEntityMapper.toDomain(entity.getUser()));
+    comment.setBody(entity.getBody());
     return comment;
   }
 
+  public List<Comment> toDomain(List<CommentEntity> commentEntities) {
+    if (commentEntities == null || commentEntities.isEmpty()) {
+      return Collections.emptyList();
+    }
+    List<Comment> commentList = new ArrayList<>(commentEntities.size());
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/database/mapper/CommentEntityMapper.java:29]: CLEAN CODE: Esto lo he mencionado en multiples PR, los nombres de la variables no deben hacer referencia a las implementaciones o tipos.

File: src/main/java/com/alkemy/ong/infrastructure/rest/mapper/ListCommentMapper.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.Collections;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@AllArgsConstructor
+public class ListCommentMapper {
+
+  CommentMapper commentMapper;
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/rest/mapper/ListCommentMapper.java:14]: ENCAPSULACION && INMUTABILIDAD

**Review by mkain0 — CHANGES_REQUESTED** [review]

**Review by mkain0 — APPROVED** [review]

**Pull request merged**